store/src/java/com/zimbra/cs/mailbox/ContactBackupThread.java

```diff
@@ -148,99 +100,71 @@ private void setContactBackupLastMailboxId(int id) {
+                sb.append(mailboxId).append(",");
+                Account account = mbox.getAccount();
+                sb.append(account.getMail()).append(",");
+                ZimbraLog.addAccountNameToContext(account.getName());
+                if (account.isFeatureContactBackupEnabled() && !account.isIsSystemAccount() && !account.isIsSystemResource() && account.isAccountStatusActive()) {
+                    OperationContext octxt = new OperationContext(account);
+                    Folder folder = getContactBackupFolder(octxt, mbox, true);
+                    if (folder != null) {
+                        createBackup(octxt, mbox, account, folder, startTime);
+                        purgeOldBackups(octxt, mbox, folder, startTime);
                     } else {
-                        ZimbraLog.contactbackup.debug("contact backup skipped: feature is disabled/account is inactive/it's a system account");
+                        success = false;
+                        ZimbraLog.contactbackup.info("contact backup folder not found for %d, continuing to next mailbox", mailboxId);
+                    }
+                    // set current mailbox id as last processed mailbox
+                    if (success) {
+                        setContactBackupLastMailboxId(mailboxId);
+                        sb.append("success");
                     }
-                } catch (Exception e) {
-                    ZimbraLog.contactbackup.warn("backup/purge failed for mailbox %d, continuing to next mailbox", mailboxId, e);
+                } else {
+                    ZimbraLog.contactbackup.debug("contact backup skipped for %d: feature is disabled/account is inactive/it's a system account", mailboxId);
+                    sb.append("skipped");
                 }
-                ZimbraLog.clearContext();
+            } catch (Exception e) {
+                ZimbraLog.contactbackup.warn("backup/purge failed for mailbox %d, continuing to next mailbox", mailboxId, e);
+                sb.append("failed");
             }
-            setContactBackupLastMailboxId(0);
-            Date endTime = new Date();
-            long diff = endTime.getTime() - startTime.getTime();
-            ZimbraLog.contactbackup.debug("finished iteration on mailboxes, iteration took %d ms", diff);
-            long sleepThreadTime = contactBackupFreq - diff; // find actual diff so that thread will start again at a same time next time
-            Date next = new Date(endTime.getTime() + sleepThreadTime);
-            setContactBackupNextIterationStartTime(next.getTime());
-            ZimbraLog.contactbackup.info("Thread finished iteration, sleeping for %d ms", sleepThreadTime);
-            sleepThread(sleepThreadTime);
+            sb.append("\n");
+            ZimbraLog.clearContext();
         }
+        setContactBackupLastMailboxId(0);
+        createReport(sb, startTime);
+        Date endTime = new Date();
```

Comment:
Can we use System.getCurrentTimeinMillis() here

Reply:
done